src/logic/schedule-exporter/workers-absence-export.logic.ts

```diff
@@ -0,0 +1,162 @@
+        }
+
+        workSheet.mergeCells(top, 1, bottom, 1);
+        workSheet.mergeCells(top, 6, bottom, 6);
+        workSheet.mergeCells(top, 7, bottom, 7);
+      }
+    });
+
+    colLens.forEach((len, id) => {
+      workSheet.getColumn(id + 1).width = len + CELL_MARGIN;
+    });
+
+    workSheet.getColumn(1).alignment = { vertical: "middle", horizontal: "left" };
+    for (let i = 1; i <= ABSENCE_HEADERS.length; i++) {
+      workSheet.getColumn(i).alignment = { vertical: "middle", horizontal: "center" };
+    }
+
+    workSheet.getRow(1).alignment = { vertical: "middle", horizontal: "center" };
+    workSheet.getRow(1).font = { bold: true };
+  }
+
+  private createWorkersInfoSection(
+    scheduleModel: ScheduleDataModel,
+    revision: PrimaryMonthRevisionDataModel
+  ): (string | number)[][] {
+    const names = Object.keys(scheduleModel.employee_info?.type);
+    const month = scheduleModel.schedule_info.month_number;
+    const year = scheduleModel.schedule_info.year;
+    const dates = scheduleModel.month_info.dates;
+    const workers: (string | number)[][] = [];
+
+    const shiftTypes = scheduleModel.shift_types;
+
+    let employeeRowIndex = 2;
+
+    workers.push(ABSENCE_HEADERS);
+    workers.push(EMPTY_ROW);
+    names.forEach((name) => {
+      const workerShifts = scheduleModel.shifts[name] as ShiftCode[];
+      const workerContractType = scheduleModel.employee_info.contractType[name];
+      const { verboseDates } = new MonthInfoLogic(month, year, dates);
+
+      let moreThanOneRow = false;
+      workerShifts.forEach((shift, index) => {
+        let daysNo = 0;
+        if (
+          !(
+            verboseDates[index].isPublicHoliday ||
+            verboseDates[index].dayOfWeek === WeekDay.SA ||
+            verboseDates[index].dayOfWeek === WeekDay.SU
+          )
+        )
+          daysNo++;
+        if (shift !== ShiftCode.W && shift !== ShiftCode.NZ && !SHIFTS[shift].isWorkingShift) {
+          employeeRowIndex++;
+          const from = scheduleModel.month_info.dates[index];
+          while (
+            index + 1 < workerShifts.length &&
+            workerShifts[index + 1] === workerShifts[index]
+          ) {
+            workerShifts[index] = ShiftCode.W;
+            index++;
+            if (
+              !(
+                verboseDates[index].isPublicHoliday ||
+                verboseDates[index].dayOfWeek === WeekDay.SA ||
+                verboseDates[index].dayOfWeek === WeekDay.SU
+              )
+            )
+              daysNo++;
+          }
+          workerShifts[index] = ShiftCode.W;
+          if (!moreThanOneRow) {
+            workers.push([
+              name,
+              SHIFTS[shift].name,
+              `${from} ${TranslationHelper.polishMonthsGenetivus[month]}`,
+              `${scheduleModel.month_info.dates[index]} ${TranslationHelper.polishMonthsGenetivus[month]}`,
+              daysNo,
+              WorkerHourInfo.calculateFreeHoursForContractType(
```

Comment:
Z tego co widzę to niestety nie działa np. Pracownik ma urlop w normalne dni, ale pomimo tego ma dalej 0 nadgodzin. Musimy jeszcze to przemyśleć i napisać algorytm do liczenia tego, ale nie będzie to już częścią tego zadania.